File: SocialPulse/SocialPulse.Core/Entities/Report.cs
﻿namespace SocialPulse.Core
{
    public class Report : BaseEntity
    {
        public string ReportReason { get; set; } = null!;

        public int ReporterId { get; set; }
        public User Reporter { get; set; } = null!;
        public int ReportedId { get; set; }
        public User Reported { get; set; } = null!;
    }
}

File: SocialPulse/UI/mobile_socialpulse/lib/models/report.dart
import 'package:json_annotation/json_annotation.dart';

part 'report.g.dart';

@JsonSerializable()
class Report{
  int? id;
  String? reportReason;
  int? reporterId;
  int? reportedId;

  Report (this.id,this.reportReason,this.reporterId,this.reportedId);

  factory Report.fromJson(Map<String, dynamic> json) =>
      _$ReportFromJson(json);

  Map<String, dynamic> toJson() => _$ReportToJson(this);
}